script/watch.js

```diff
@@ -0,0 +1,170 @@
+// babel-register only looks for environment variables
+process.env.BABEL_ENV = process.env.BABEL_ENV || 'test';
+// use babel-register to compile files on the fly
+require('babel-register');
+
+// require mocha setup files
+require('../test/util/mocha-setup.js');
+require('../test/util/enzyme-setup.js');
+
+const { Writable } = require('stream');
+const chokidar = require('chokidar');
+const CLIEngine = require('eslint').CLIEngine;
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const glob = util.promisify(require('glob'));
+
+const eslint = new CLIEngine();
+const formatter = eslint.getFormatter('stylish');
+const sasslinter = require('sass-lint');
+const Mocha = require('mocha');
+
+const chalk = require('chalk');
+const path = require('path');
+const _ = require('lodash');
+
+// no arguments, run unit tests
+// any arguments provited then tests run must be explicitly provided
+let options;
+if (process.argv.length === 2) {
+  options= {
+    eslint: false,
+    sasslint: false,
+    showErrors: false,
+    unit: true
+  };
+} else {
+  options = {
+    eslint: process.argv.includes('eslint'),
+    sasslint: process.argv.includes('sasslint'),
+    showErrors: process.argv.includes('showErrors'),
+    unit: process.argv.includes('unit')
+  };
+}
+
+const watchArray = [
+  './src/**/*.jsx',
+  './src/**/*.js',
+  './test/**/*.unit.spec.*'
+];
+
+if (options.sassLint) {
+  watchArray.push('./src/**/*.scss');
+}
+
+const srcWatcher = chokidar.watch(watchArray);
+
+async function runTests() {
+  const groupedFiles = groupFilesByTest(await getChangedFiles());
+
+  if (options.unit) {
+    const unitTests = groupedFiles.mocha.length > 0 ?
+      groupedFiles.mocha :
+      await glob('test/**/*.unit.spec.js?(x)');
+    console.log(`Running Mocha with ${unitTests.length} Files\n`);
+    // mocha requires flushing the require cache of all js files (tests and imports)
+    groupedFiles.eslint.forEach(file => delete require.cache[file]);
```

Comment:
This is something I don't fully understand. I think because `mocha` requires the test and SUT files, then these files have to be flushed out of the node `require` cache so that `mocha` can run again. Without doing this, `mocha` only runs once or doesn't run with updated files.